### DISCUSSION TITLE
Type definitions for machine.children?

### DISCUSSION THREAD

**Discussion by bstro** [statelyai/xstate#3473] [Q&A] [answered]
Hi there,

Generally speaking, when using typegen, should I expect `state.children.*` to be typed? I've put together a state machine and realized that when it came time to actually consume it, despite having the typegen definitions generated, that after invoking other machines on my parent machine, that any machines on `state.children.*` are not typed with any degree of specificity.

For example, my root machine invokes a `selection` machine. From react, upon `use`ing that machine, my `send` method is untyped:
<img width="672" alt="image" src="https://user-images.githubusercontent.com/627290/181164702-e47a467e-32f2-4aff-bf9d-1f2e301f4c89.png">

I have the same issue with `getSnapshot`. There's no types specific to my machine when I `invoke` (I have experienced more luck with spawning machines and storing them in context).

**Comment by Andarist** [comment, marked as answer]
> Generally speaking, when using typegen, should I expect state.children.* to be typed? 

Not right now. But we have plans to improve this situation.

**Reply by bstro** [reply]
Cool! Is that work/planning tracked anywhere on Github?

**Reply by Andarist** [reply]
Not yet - we are going to work on this as part of the ongoing work towards the next major version.
